webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/db/DB.scala

```diff
@@ -309,21 +310,23 @@ final case class SerializedRules[T](
               at.techniqueName.value,
               technique.name,
               Opt(technique.description),
-              directive.techniqueVersion.toString,
+              directive.techniqueVersion.serialize,
               DateTime.now(), None )
     }
 
+    // TODO: this is only used for history, no version here
     def fromSerializedRule(
         rule : SerializedRules[Long]
       , ruleTargets : Seq[SerializedRuleGroups]
       , directives : Seq[SerializedRuleDirectives]
     ) : Rule = {
       Rule (
           RuleId(rule.ruleId)
+        , None// TODO: we should perhaps be able to get ride of these table.
         , rule.name
         , RuleCategoryId(rule.categoryId.getOrElse("rootRuleCategory")) // this is not really useful as RuleCategory are not really serialized
         , ruleTargets.flatMap(x => RuleTarget.unser(x.targetSerialisation)).toSet
-        , directives.map(x => DirectiveId(x.directiveId)).toSet
+        , directives.map(x => DirectiveRId(DirectiveId(x.directiveId))).toSet
```

Comment:
I think we should store and get back the directive id with revision from de table if we can, or else the directive here won't be the same

Reply:
same as above: I would love to delete histororization repo.